changelog.d/20230106_190620_regis_hooks_api.md

```diff
@@ -0,0 +1 @@
+- 💥[Feature] Simplify the hooks API. Plugin developers who were previously using `hooks.actions`, `hooks.filters` or `hooks.contexts` should now import these modules explicitely. (by @regisb)
```

Comment:
I'm not sure what the changelog syntax allows, but what do think about enumerating the breaking changes here?

```suggestion
- 💥[Feature] Simplify the hooks API. The modules `tutor.hooks.actions`, `tutor.hooks.filters`, and `tutor.hooks.contexts` are no longer part of the API.
  * Instead of `tutor.hooks.actions.get("some:action")` , use `tutor.hooks.Action.SOME_ACTION` or `tutor.core.hooks.Actions.get("some:action")`. The former is highly preferred for all predefined Tutor Actions, as its usage can be checked by pylint and mypy.
  * Instead of `tutor.hooks.filters.get("some:filter")`, use `tutor.hooks.Filter.SOME_FILTER` or `tutor.core.hooks.Filter.get("some:filter")`. The former is highly preferred for all predefined Tutor , as its usage can be checked by pylint and mypy.
  * Instead of `tutor.hooks.actions.add("some:action")`, use `tutor.hooks.Action.SOME_ACTION.add()`. The same applies to the `do` method.
  * Instead of `tutor.hooks.actions.add("some:action")`, use `tutor.hooks.Action.SOME_ACTION.add()`. The same applies to the `add_item`, `add_items`, `apply`, and `iterate` methods.
  * ... and a couple more bullet points about changes to context-related functions
  (by @regisb)    
```